xonsh/main.py

```diff
@@ -163,6 +163,11 @@ def premain(argv=None):
         # and positional arguments passed before the path to the script-file are
         # ignored.
         args.args = arguments[file_index+1:]
+    elif not args.args and other and other[0].startswith('-'):
```

Comment:
I think that the problem here is that on line 160, you need to reassign `other`. We are using the old other, and we need to use the new other.